R/rowWelchTests.R

```diff
@@ -66,22 +66,28 @@ rowWelchTests <- function(X, categ,
   nX <- as.matrix(colSums(categ0))
   nY <- as.matrix(colSums(categ1))
 
-  sumX <- X %*% categ0
-  sumY <- X %*% categ1
-  XX <- X * X
-  sum2X <- XX %*% categ0
-  sum2Y <- XX %*% categ1
-
-  mX <- divide_cols(sumX, nX)
-  mY <- divide_cols(sumY, nY)
-
-  num <- sum2X - divide_cols(sumX^2, nX)
-  num <- abs(num) # fix rare corner cases where num ~= -1e-15
-  sX <- sqrt(divide_cols(num, nX - 1))
-
-  num <- sum2Y - divide_cols(sumY^2, nY)
-  num <- abs(num) # fix rare corner cases where num ~= -1e-15
-  sY <- sqrt(divide_cols(num, nY - 1))
+  sumX <- X %*% categ0  
+  sum2X <- (X * X) %*% categ0  
+  mX <- divide_cols(sumX, nX) 
+  
+  num <- sum2X - divide_cols(sumX^2, nX)  
+  rm(sumX)  
+  rm(sum2X)  
+  gc()
+  num <- abs(num) # fix rare corner cases where num ~= -1e-15  
+  sX <- sqrt(divide_cols(num, nX - 1)) 
+  
+  
+  sumY <- X %*% categ1  
+  sum2Y <- (X * X) %*% categ1  
```

Comment:
Here we are calculating X *  X on the fly a second time. Why not storing the result as before ? 

(I don't think we will gain much in time because X is n x m and typically n << B so B*m is the limiting factor, both in time and space. However this also means that not storing X * X will not save much memory)